### ISSUE TITLE
missing reserve() in core/string.h

### ISSUE THREAD

**Issue by ppiecuch** [nardo/kt#1]
Seems like this is missing in string.h - maybe you did not commit new version of core?

Regards
Pawel

**Comment by nardo** [comment]
Good call. New version of core committed. See if that works for you.

On Tue, Aug 11, 2015 at 1:04 PM ppiecuch <notifications@github.com> wrote:

> Seems like this is missing in string.h - maybe you did not commit new
> version of core?
>
> Regards
> Pawel
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nardo/kt/issues/1>.
>
-- 
Sent from mobile. Please excuse typos, brevity and nonsensical expressions.

**Comment by ppiecuch** [comment]
Almost

Converting from 0x1000125a8 to 0x1000125a8
Converting from 0x100012748 to 0x1000125a8
10
Converting from 0x1000125a8 to 0x1000125a8
hello443211911111410810033
test1_out(17393,0x7fff7c2a2310) malloc: *** error for object 0x100200710: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
Process 17393 stopped



Backtrace:
(lldb) bt
* thread #1: tid = 0x683cc, 0x00007fff91fa0bc0 libsystem_malloc.dylib`malloc_error_break, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1 2.1 3.1
  * frame #0: 0x00007fff91fa0bc0 libsystem_malloc.dylib`malloc_error_break
    frame #1: 0x00007fff91fa2028 libsystem_malloc.dylib`free + 324
    frame #2: 0x0000000100003055 test1_out`core::memory_deallocate(ptr=0x0000000100200710) + 21 at memory_functions.h:12
    frame #3: 0x0000000100003029 test1_out`core::string::_dec_ref(this=0x00007fff5fbff4f0) + 89 at string.h:23
    frame #4: 0x000000010000e565 test1_out`core::string::~string(this=0x00007fff5fbff4f0) + 21 at string.h:34
    frame #5: 0x0000000100001175 test1_out`core::string::~string(this=0x00007fff5fbff4f0) + 21 at string.h:33
    frame #6: 0x00000001000012c1 test1_out`core::program::main() + 225 at test1_output.cpp:86
    frame #7: 0x0000000100000df0 test1_out`main(argc=1, argv=0x00007fff5fbff5a8) + 32 at test1_output.cpp:100
    frame #8: 0x00007fff969545fd libdyld.dylib`start + 1
    frame #9: 0x00007fff969545fd libdyld.dylib`start + 1

**Comment by nardo** [comment]
Yes, that is as far as I've gotten -- I haven't tracked down the memory
bug, and the string is concatenating integers rather than character data.
On Wed, Aug 12, 2015 at 1:14 PM ppiecuch <notifications@github.com> wrote:

> Almost
>
> Converting from 0x1000125a8 to 0x1000125a8
> Converting from 0x100012748 to 0x1000125a8
> 10
> Converting from 0x1000125a8 to 0x1000125a8
> hello443211911111410810033
> test1_out(17393,0x7fff7c2a2310) malloc: *** error for object 0x100200710:
> pointer being freed was not allocated
> *** set a breakpoint in malloc_error_break to debug
> Process 17393 stopped
>
> Backtrace:
> (lldb) bt
>
>    - thread #1 <https://github.com/nardo/kt/issues/1>: tid = 0x683cc,
>    0x00007fff91fa0bc0 libsystem_malloc.dylib`malloc_error_break, queue =
>    'com.apple.main-thread', stop reason = breakpoint 1.1 2.1 3.1
>       - frame #0: 0x00007fff91fa0bc0 libsystem_malloc.dylibmalloc_error_break
>       frame #1: 0x00007fff91fa2028 libsystem_malloc.dylibfree + 324 frame
>       #2: 0x0000000100003055 test1_outcore::memory_deallocate(ptr=0x0000000100200710)
>       + 21 at memory_functions.h:12 frame #3: 0x0000000100003029 test1_outcore::string::_dec_ref(this=0x00007fff5fbff4f0)
>       + 89 at string.h:23 frame #4: 0x000000010000e565 test1_outcore::string::~string(this=0x00007fff5fbff4f0)
>       + 21 at string.h:34 frame #5: 0x0000000100001175 test1_outcore::string::~string(this=0x00007fff5fbff4f0)
>       + 21 at string.h:33 frame #6: 0x00000001000012c1 test1_outcore::program::main()
>       + 225 at test1_output.cpp:86 frame #7: 0x0000000100000df0 test1_outmain(argc=1,
>       argv=0x00007fff5fbff5a8) + 32 at test1_output.cpp:100 frame #8:
>       0x00007fff969545fd libdyld.dylibstart + 1 frame #9:
>       0x00007fff969545fd libdyld.dylibstart + 1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nardo/kt/issues/1#issuecomment-130431976>.
>
-- 
Sent from mobile. Please excuse typos, brevity and nonsensical expressions.
